sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -191,14 +193,25 @@ def query_factory() -> Query:
                     elif getattr(df.dtypes[column], "tz", None) is not None:  # type: ignore
                         # MSSQL requires a colon in the offset (+00:00) so we use isoformat() instead of strftime()
                         df[column] = pd.to_datetime(df[column]).map(lambda x: x.isoformat(" "))  # type: ignore
+
+                        # bulk_copy() doesn't work with TZ timestamp, so load into string column and cast to
+                        # timestamp in SELECT statement
+                        columns_to_types_tztext = columns_to_types.copy()
+                        columns_to_types_tztext[column] = exp.DataType.build("TEXT")
                     else:  # type: ignore
                         df[column] = pd.to_datetime(df[column]).dt.strftime("%Y-%m-%d %H:%M:%S.%f")  # type: ignore
 
-            self.create_table(temp_table, columns_to_types)
+            self.create_table(
+                temp_table, columns_to_types_tztext if columns_to_types_tztext else columns_to_types
```

Comment:
[Nit] More common pattern in this codebase: `columns_to_types_tztext or columns_to_types`